agent/engine/daemonmanager/daemon_manager_linux.go

```diff
@@ -79,16 +90,23 @@ func (dm *daemonManager) CreateDaemonTask() (*apitask.Task, error) {
 	}
 
 	for _, mp := range dm.managedDaemon.GetMountPoints() {
-		err := mkdirAllAndChown(mp.SourceVolumeHostPath, daemonMountPermission, daemonUID, os.Getegid())
-		if err != nil {
+		if err := mkdirAllAndChown(mp.SourceVolumeHostPath, daemonMountPermission, daemonUID, os.Getegid()); err != nil {
 			return nil, err
 		}
 		var bindOptions = dockermount.BindOptions{}
 
 		if mp.PropagationShared {
 			// https://github.com/moby/moby/blob/master/api/types/mount/mount.go#L52
```

Comment:
I know this is not a part of this PR but we should use permalinks so that it's tied to a commit ID instead of branch name which points to the tip of the branch.